samples/subsys/usb/hid-mouse/src/main.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2018 qianfan Zhao
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <board.h>
+#include <device.h>
+#include <gpio.h>
+
+#include <usb/usb_device.h>
+#include <usb/class/usb_hid.h>
+
+/* change this to use another GPIO port */
+#ifdef SW0_GPIO_NAME
+#define PORT	SW0_GPIO_NAME
+#else
+#error SW0_GPIO_NAME needs to be set in board.h
+#endif
+
+/* change this to use another GPIO pin */
+#ifdef SW0_GPIO_PIN
+#define PIN     SW0_GPIO_PIN
+#else
+#error SW0_GPIO_PIN needs to be set in board.h
+#endif
+
+static const u8_t hid_report_desc[] = {
+	HID_GI_USAGE_PAGE, USAGE_GEN_DESKTOP,
+	/* USAGE_PAGE (Generic Desktop)				05 01 */
+	HID_LI_USAGE, USAGE_GEN_DESKTOP_MOUSE,
+	/* USAGE (Mouse)					09 02 */
+		HID_MI_COLLECTION, COLLECTION_APPLICATION,
+		/* COLLECTION (Application)			A1 01 */
+		HID_LI_USAGE, USAGE_GEN_DESKTOP_POINTER,
+		/* 	USAGE (Pointer)				09 01 */
+		HID_MI_COLLECTION, COLLECTION_PHYSICAL,
+		/* 	COLLECTION (Physical)			A1 00 */
+				HID_GI_USAGE_PAGE, USAGE_GEN_BUTTON,
+		/* 		USAGE_PAGE (Button)		05 09 */
+				HID_LI_USAGE_MIN(1), 0x01,
+		/* 		USAGE_MINIMUM (Button 1)	19 01 */
+				HID_LI_USAGE_MAX(1), 0x03,
+		/* 		USAGE_MAXIMUM (Button 3)	29 03 */
+				HID_GI_LOGICAL_MIN(1), 0,
+		/* 		LOGICAL_MINIMUM (0)		15 00 */
+				HID_GI_LOGICAL_MAX(1), 1,
+		/* 		LOGICAL_MAXIMUM (1)		25 01 */
+				HID_GI_REPORT_COUNT, 3,
+		/* 		REPORT_COUNT (3)		95 03 */
+				HID_GI_REPORT_SIZE, 1,
+		/* 		REPORT_SIZE (1)			75 01 */
+				HID_MI_INPUT, 0x02,
+		/* 		INPUT (Data,Var,Abs)		81 02 */
+				HID_GI_REPORT_COUNT, 1,
+		/* 		REPORT_COUNT (1)		95 01 */
+				HID_GI_REPORT_SIZE, 5,
+		/* 		REPORT_SIZE (5)			75 05 */
+				HID_MI_INPUT, 0x01,
+		/* 		INPUT (Cnst,Ary,Abs)		81 01 */
+				HID_GI_USAGE_PAGE, USAGE_GEN_DESKTOP,
+		/* 		USAGE_PAGE (Generic Desktop)	05 01 */
+				HID_LI_USAGE, USAGE_GEN_DESKTOP_X,
+		/* 		USAGE (X)			09 30 */
+				HID_LI_USAGE, USAGE_GEN_DESKTOP_Y,
+		/* 		USAGE (Y)			09 31 */
+				HID_LI_USAGE, USAGE_GEN_DESKTOP_WHEEL,
```

Comment:
If this emulates a mouse with just one button, then there is no need to advertise wheel events.

Reply:
I think its is necessary. Yes I haven't report mouse move, but if someone want, change the 4bytes reported to usb host will achieve easier.